lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -154,31 +181,67 @@ bool ArrayAllocation::mapInitializationStores() {
 /// allocated.
 bool ArrayAllocation::isInitializationWithKnownElements() {
   ArraySemanticsCall Uninitialized(Alloc, "array.uninitialized");
-  if (Uninitialized &&
-      (ArrayValue = Uninitialized.getArrayValue()) &&
-      (ElementBuffer = Uninitialized.getArrayElementStoragePointer()))
+  if (Uninitialized && (ArrayValue = Uninitialized.getArrayValue()) &&
```

Comment:
Can you invert this if statement? It is always better to make early exits the really simple case (in this case the return false).